communication/mavlink_waypoint_handler.cpp

```diff
@@ -114,34 +106,20 @@ void Mavlink_waypoint_handler::send_waypoint(Mavlink_waypoint_handler* waypoint_
         if (((uint8_t)packet.target_system == (uint8_t)sysid)
-                && ((uint8_t)packet.target_component == (uint8_t)MAV_COMP_ID_MISSIONPLANNER))
+                && ((uint8_t)packet.target_component == (uint8_t)MAV_COMP_ID_MISSIONPLANNER
+                || (uint8_t)packet.target_component == 50)) // target_component = 50 is sent by dronekit
         {
             waypoint_handler->sending_waypoint_num_ = packet.seq;
             if (waypoint_handler->sending_waypoint_num_ < waypoint_handler->waypoint_count_)
             {
-                //  Prototype of the function "mavlink_msg_mission_item_send" found in mavlink_msg_mission_item.h :
-                // mavlink_msg_mission_item_send (  mavlink_channel_t chan, uint8_t target_system, uint8_t target_component, uint16_t seq,
-                //                                  uint8_t frame, uint16_t command, uint8_t current, uint8_t autocontinue, float param1,
-                //                                  float param2, float param3, float param4, float x, float y, float z)
-                mavlink_message_t _msg;
-                mavlink_msg_mission_item_pack(sysid,
-                                              waypoint_handler->mavlink_stream_.compid(),
-                                              &_msg,
-                                              msg->sysid,
-                                              msg->compid,
-                                              packet.seq,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].frame,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].command,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].current,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].autocontinue,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param1,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param2,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param3,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param4,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].x,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].y,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].z);
-                waypoint_handler->mavlink_stream_.send(&_msg);
+                uint8_t isCurrent = 0;
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/2e7b08bcde000e4cbcf2f549169f46eeb1cd1ac9